File: 83_remove_duplicates_from_sorted_list/deleteDuplicates.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        
        tmp = head
        
        while tmp and tmp.next:
            if tmp.val == tmp.next.val:
                tmp.next = tmp.next.next
            else:
                tmp = tmp.next

        return head

File: 83_remove_duplicates_from_sorted_list/deleteDuplicates.cpp
#include <iostream>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
	public:
    	ListNode* deleteDuplicates(ListNode* head) {
            ListNode* tmp = head;
            while(tmp && tmp->next) {
                if(tmp->val == tmp->next->val) {
                    ListNode* n = tmp -> next -> next;
                    delete tmp -> next;
                    tmp -> next = n;
                }
                else {
                    tmp = tmp -> next;
                }
            }

            return head;
    	}
};
